roxie/ccd/ccdfile.cpp

```diff
@@ -756,12 +756,15 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
             splitFilename(targetFilename, &destPath, &destPath, &prevTempFile, &prevTempFile);
             prevTempFile.append("*.$$$");
             Owned<IFile> dirf = createIFile(destPath.str());
-            Owned<IDirectoryIterator> iter = dirf->directoryFiles(prevTempFile.str(),false,false);
-            ForEach(*iter)
+            if (dirf)
```

Comment:
@richardkchapman - this is the fix for the failing CcdFileTest regression.

It was failing because dirf was null, because destPath was "".
Previously (prior to HPCC-19233, createIFile was inconsistent and return null when filename was null, but returned an CFile object with probably undefined behaviour for "".

I think it's correct (createIFile), but do wonder if other code like this elsewhere that was relying on createIFile("") being valid.

We can fix them in similar fashion like this, or could allow createFile("") or createIFile(nullptr) to return a proper null implementation.. what do you think?

Reply:
You could argue that "" for a directory should be treated the same as "." (which seems to be what this is assuming). When splitting a filename into directory and file parts it won't be that unusual to end up with the directory part empty.

Reply:
I'd be worried if it's changed so that createIFile("" or null) as equivalent to  "." , it may similarly lead to unexpected outcomes.

e.g. the code in ccdfile will start deleting all files in the current directory(!). Which up to now it will not have done.
I suspect this code thought there'd be a temp directory prefix such that destPath wouldn't be empty.